visitors/cli.go

```diff
@@ -0,0 +1,56 @@
+package visitors
+
+import (
+	"fmt"
+
+	"github.com/linuxboot/fiano/uefi"
+)
+
+var visitorRegistry = map[string]visitorEntry{}
+
+type visitorEntry struct {
+	numArgs       int
+	createVisitor func([]string) (uefi.Visitor, error)
+}
+
+func RegisterCLI(name string, numArgs int, createVisitor func([]string) (uefi.Visitor, error)) {
+	if _, ok := visitorRegistry[name]; ok {
+		panic(fmt.Sprintf("two visitors registered the same name: '%s'", name))
+	}
+	visitorRegistry[name] = visitorEntry{
+		numArgs:       numArgs,
+		createVisitor: createVisitor,
+	}
+}
+
+// TODO: display some type of help message
+func ParseCLI(args []string) ([]uefi.Visitor, error) {
+	visitors := []uefi.Visitor{}
+	for i := 0; i < len(args); i++ {
```

Comment:
Instead of just using i, can we slide across args for readability? E.g.

```go
for len(args) > 0 {
  o, ok := visitorRegistry[args[0]]
  ..
  if o.numArgs < len(args) {
    too few args
  }
  cmdArgs := args[0:o.numArgs]
  ...
  args = args[o.numArgs:]
}
```

this won't affect the caller, since we are only modifying the slice header.

Reply:
Thanks, I'll do this!

Reply:
Done!